agent/uiserver/uiserver.go

```diff
@@ -92,20 +92,20 @@ func (h *Handler) ReloadConfig(newCfg *config.RuntimeConfig) error {
 func (h *Handler) handleIndex() (http.Handler, error) {
 	cfg := h.getRuntimeConfig()
 
-	var fsys http.FileSystem
+	var fsys fs.FS
 	if cfg.UIConfig.Dir == "" {
 		// strip the dist/ prefix
-		sub, err := fs.Sub(assets, "dist")
+		sub, err := fs.Sub(dist, "dist")
 		if err != nil {
 			return nil, err
 		}
-		fsys = http.FS(sub)
+		fsys = sub
 	} else {
-		fsys = http.Dir(cfg.UIConfig.Dir)
+		fsys = os.DirFS(cfg.UIConfig.Dir)
 	}
 
 	// Render a new index.html with the new config values ready to serve.
-	buf, info, err := h.renderIndex(cfg, fsys)
+	buf, err := h.renderIndexFile(cfg, fsys)
```

Comment:
we were passing around `[]bytes` and `FileInfo` to reconstruct a file later; refactored to do that work in this func and pass pointer to `fs.File` around